### PR TITLE
percol 0.2.1 (new formula)

### PR DESCRIPTION
**Pull request by zmwangx**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

A word on the test block: I can't do much better than a version test due to the interactive nature of `percol`.

### CODE REVIEW

File: Formula/percol.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class Percol < Formula
+  desc "Interactive grep tool"
+  homepage "https://github.com/mooz/percol"
+  url "https://github.com/mooz/percol/archive/v0.2.1.tar.gz"
+  sha256 "75056ba1fe190ae4c728e68df963c0e7d19bfe5a85649e51ae4193d4011042f9"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  resource "six" do
+    url "https://pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz"
```
Comment (ilovezfs) [inline Formula/percol.rb]: current preferred url is https://files.pythonhosted.org/packages/b3/b2/238e2590826bfdd113244a40d9d3eb26918bd798fc187e2360a8367068db/six-1.10.0.tar.gz

File: Formula/percol.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a"
+  end
+
+  resource "cmigemo" do
+    url "https://pypi.python.org/packages/source/c/cmigemo/cmigemo-0.1.6.tar.gz"
```
Comment (ilovezfs) [inline Formula/percol.rb]: https://files.pythonhosted.org/packages/2f/e4/374df50b655e36139334046f898469bf5e2d7600e1e638f29baf05b14b72/cmigemo-0.1.6.tar.gz

**Comment by ilovezfs** [comment]
@zmwangx You can do something like this:
```
  test do
    (testpath/"expect_script").write <<-EOS.undent
      spawn #{bin}/percol --auto-match --query=_windowserver psaux.txt
      expect "QUERY> _windowserver"
      send "\r\003"
    EOS
    shell_output("/bin/ps aux >psaux.txt")
    output = shell_output("/usr/bin/expect -f expect_script")
    assert_match "CoreGraphics.framework", output
  end
```

File: Formula/percol.rb (hunk 1 above)
Comment (zmwangx) [inline Formula/percol.rb]: @ilovezfs Why not https://pypi.python.org/packages/b3/b2/238e2590826bfdd113244a40d9d3eb26918bd798fc187e2360a8367068db/six-1.10.0.tar.gz?

Comment (ilovezfs) [inline Formula/percol.rb]: @zmwangx https://github.com/tdsmith/homebrew-pypi-poet is generally the way of generating these blocks and @tdsmith has recently switched over the URLs to the pythonhosted.org ones.

**Comment by zmwangx** [comment]
Good call on using expect. I updated the test to grep Homebrew's README, which is more cross platform.

**Comment by ilovezfs** [comment]
Looks good. Did you want to add `head` too?

**Comment by zmwangx** [comment]
Well, I'm generally hesitant to add head, because it's really a horrible experience when it comes to updates. But it doesn't hurt to include it in the formula, so I'll add it if you say so.

**Comment by ilovezfs** [comment]
Makes no difference to me. Leave out it if you'd like.

**Comment by zmwangx** [comment]
Added anyway.

**Comment by MikeMcQuaid** [comment]
Thanks again @zmwangx!

File: Formula/percol.rb (hunk 1 above)
Comment (tdsmith) [inline Formula/percol.rb]: FWIW I do not prefer one URL over the other; homebrew-pypi-poet switched to the new Warehouse PyPI API endpoint in order to get hashes over the wire and those happen to be the URLs Warehouse is returning

Comment (zmwangx) [inline Formula/percol.rb]: Fair enough. (So Warehouse's API is actually usable at this point?)

**Pull request closed without merging**